### PR TITLE
Bump to go1.22

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module github.com/transparency-dev/formats
 
-go 1.21.0
+go 1.22.0
 
-toolchain go1.22.3
+toolchain go1.22.5
```
Comment (roger2hk) [inline go.mod:5]:
```suggestion
toolchain go1.22.7
```

File: .github/workflows/go_test.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.21.x, 1.22.x]
+        go-version: [1.22.x]
```
Comment (roger2hk) [inline .github/workflows/go_test.yml:9]:
```suggestion
        go-version: [1.22.x, 1.23.x]
```

**Review by roger2hk — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (AlCutter) [reply go.mod]: Unfortunately, we can't bump to 1.22.7 yet as TamaGo is only at 1.22.6 currently.

Comment (roger2hk) [reply go.mod]: Ah right, it will take a few days to catch up.

File: .github/workflows/golangci-lint.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
     steps:
       - uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
         with:
-          go-version: 1.21
+          go-version: 1.22
```
Comment (roger2hk) [inline .github/workflows/golangci-lint.yml:16]:
```suggestion
          go-version-file: go.mod
```

**Review by roger2hk — APPROVED** [review]

**Review by roger2hk — APPROVED** [review]

**Pull request merged**